core/src/main/java/com/opensymphony/xwork2/conversion/impl/StringConverter.java

```diff
@@ -65,8 +65,17 @@ protected String convertToString(Locale locale, Object value) {
         if (Number.class.isInstance(value)) {
             NumberFormat format = NumberFormat.getNumberInstance(locale);
             format.setGroupingUsed(false);
-            if (Double.class.isInstance(value) || BigDecimal.class.isInstance(value)) {
+            if (BigDecimal.class.isInstance(value) || Double.class.isInstance(value) || Float.class.isInstance(value)) {
                 format.setMinimumFractionDigits(1);
+                if (BigDecimal.class.isInstance(value)) {
+                    format.setMaximumFractionDigits(Integer.MAX_VALUE);
+                }
+                else if (Double.class.isInstance(value)) {
+                    format.setMaximumFractionDigits(325);   //double MIN_VALUE
```

Comment:
@yasserzamani Can you explain where these numbers coming from? How do you find them?

Reply:
For BigDecimal theoretically there is no limit for count of fraction digits [1] but technically setMaximumFractionDigits cannot get bigger than Integer.MAX_VALUE.

For Double and Float, I converted `Double.MIN_VALUE` and `Float.MIN_VALUE` to a not scientific notated string by java 7 8 and 9 and found that numbers.

[1] http://download.oracle.com/javase/6/docs/api/java/math/BigDecimal.html

Reply:
@aleksandr-m what do you think about keeping max integer for all instead:

```java
if (BigDecimal.class.isInstance(value) || 
    Double.class.isInstance(value) || Float.class.isInstance(value)) {
          format.setMinimumFractionDigits(1);
          format.setMaximumFractionDigits(Integer.MAX_VALUE);
}
```

Reply:
I think that's ok

Reply:
👍  Fixed